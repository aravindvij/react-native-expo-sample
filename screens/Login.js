import React, { Component } from "react";
import {
  View,
  Text,
  ScrollView,
  StyleSheet,
  KeyboardAvoidingView,
  Button,
  TouchableHighlight,
  ActivityIndicator,
  FlatList
} from "react-native";
import InputField from "../components/form/inputField";
import NextArrowButton from "../components/buttons/NextArrowButton";
import Icon from "react-native-vector-icons/FontAwesome";

// const fetch = require("node-fetch");

import * as Localization from 'expo-localization';
import i18n from 'i18n-js';

i18n.translations = {
  en: { welcome: 'Username', name: 'Charlie' },
  ja: { welcome: 'こんにちは' },
};

// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;
// When a value is missing from a language it'll fallback to another language with the key present.
i18n.fallbacks = true;

export default class Login extends Component {
  render() {
    return (
      <KeyboardAvoidingView style={styles.wrapper} behavior="padding">
        <View style={styles.scrollViewWrapper}>
          <ScrollView style={styles.scrollView}>
            <Text style={styles.loginHeader}>Login</Text>
            <InputField
              labelTextSize={16}
              customStyle={{ marginBottom: 10 }}
              inputType="email"
              labelText={i18n.t('welcome')}
            />
            <InputField
              labelTextSize={16}
              inputType="password"
              labelText="Password"
            />
          </ScrollView>
          {/* <NextArrowButton/> */}
        </View>
        <View style={styles.buttonWrapper}>
          <TouchableHighlight style={[styles.button]}>
            <Button
              color="gray"
              title=">"
              onPress={() => this.props.navigation.navigate("Home")}
            ></Button>
          </TouchableHighlight>
        </View>
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  wrapper: {
    display: "flex",
    flex: 1,
    backgroundColor: "white"
  },
  scrollViewWrapper: {
    marginTop: 70,
    flex: 1
  },
  avoidView: {
    paddingLeft: 30,
    paddingRight: 30,
    paddingTop: 20,
    flex: 1
  },
  loginHeader: {
    fontSize: 28,
    color: "black",
    fontWeight: "300",
    marginBottom: 40,
    margin: 10
  },
  button: {
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 50,
    width: 60,
    height: 60,
    backgroundColor: "lightgray"
  },
  buttonWrapper: {
    alignItems: "flex-end",
    right: 20,
    bottom: 20,
    paddingTop: 0
  }
});

// class Login extends Component {
//   constructor(props) {
//     super(props);
//     this.state = { isLoading: true };
//   }
//   componentDidMount() {
//     return fetch("https://facebook.github.io/react-native/movies.json")
//       .then(response => response.json())
//       .then(responseJson => {
//         this.setState(
//           {
//             isLoading: false,
//             dataSource: responseJson.movies
//           },
//           function() {}
//         );
//       })
//       .catch(error => {
//         console.error(error);
//       });
//   }

//   render() {
//     return (
//       <View>
//         <Text>{this.state.data.body}</Text>
//       </View>
//     );
//     if (this.state.isLoading) {
//       return (
//         <View style={{ flex: 1, padding: 20 }}>
//           <ActivityIndicator />
//         </View>
//       );
//     }
//     return (
//       <View style={{ flex: 1, paddingTop: 20 }}>
//         <FlatList
//           data={this.state.dataSource}
//           renderItem={({ item }) => (
//             <Text>
//               {item.title}, {item.releaseYear}
//             </Text>
//           )}
//           keyExtractor={({ id }, index) => id}
//         />
//       </View>
//     );
//   }
// }
// export default Login;
