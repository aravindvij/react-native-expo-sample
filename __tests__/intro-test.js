import React from 'react';
import Login from '../screens/Login';

import renderer from 'react-test-renderer';

jest.useFakeTimers()

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});