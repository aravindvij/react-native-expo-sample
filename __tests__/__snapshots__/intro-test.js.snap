// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "display": "flex",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 70,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 28,
              "fontWeight": "300",
              "margin": 10,
              "marginBottom": 40,
            }
          }
        >
          Login
        </Text>
        <View
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "display": "flex",
                "marginLeft": 10,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            E-mail
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderBottomColor": "transparent",
                  "color": "black",
                },
                Object {
                  "borderBottomColor": "black",
                  "borderBottomWidth": 0.5,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              undefined,
              Object {
                "display": "flex",
                "marginLeft": 10,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Password
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontWeight": "600",
                }
              }
            >
              Show
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "borderBottomColor": "transparent",
                  "color": "black",
                },
                Object {
                  "borderBottomColor": "black",
                  "borderBottomWidth": 0.5,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "paddingTop": 0,
        "right": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "lightgray",
            "borderRadius": 50,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "gray",
                },
              ]
            }
          >
            &gt;
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
